### DISCUSSION TITLE
Changes to presets for existing users

### DISCUSSION THREAD

**Discussion by mxmou** [ScratchAddons/ScratchAddons#3855] [Discussion]
There are currently two open PRs that change editor dark mode presets: #3710 and #3839. Users who are already using those presets should get the updated colors, and those who selected the preset and then changed some colors would ideally get them as well. There are multiple ways to do that:
* Update any setting that has the same value as the old preset, even if no other setting matches the preset.
* Check if all affected settings have the same values as in the old version of the preset. For example, if a TurboWarp Dark user changed the border color, the accent background wouldn't be changed either.
* Compare all settings with the preset values and update them if at least *n* (half?) match the preset.

Which one should be used?

**Comment by Dhuls-Scratch** [comment]
IMO, the first one could cause things to look weird. the second option is my preferred one, although the third one is fine.

**Comment by Samq64** [comment]
I think it should stay the way it is right now. No colours get changed when updating and the user can still get the new ones very easily since it's a preset.

**Reply by mxmou** [reply]
I don't think that would work without a way to notify the user about the updated preset.

**Reply by WorldLanguages** [reply]
#3811 ?

**Reply by mxmou** [reply]
I don't think that's enough because it's on the settings page.

**Comment by WorldLanguages** [comment]
> Check if all affected settings have the same values as in the old version of the preset. For example, if a TurboWarp Dark user changed the border color, the accent background wouldn't be changed either.

I think we should go for this one. It feels weird to force settings to change if a user is not on "default settings" of a preset. Taking an extra step, no matter how small, to have the settings similar to a preset but not exactly the same, is customizing the preset, and changing the settings without consent doesn't sound good. Most users that just chose a preset and moved on wouldn't care if their preset got "updated".
If we get feedback about users who liked the old version of the presets, we can then consider adding `(legacy)` presets.

**Reply by WorldLanguages** [reply]
@mxmou Tell me if you think this is fine.

**Reply by cobaltt7** [reply]
> If we get feedback about users who liked the old version of the presets, we can then consider adding `(legacy)` presets.

How different are the updated versions? I didn't think the changes were that noticeable (just from reading prs, not from eating them)

**Reply by mxmou** [reply]
> > If we get feedback about users who liked the old version of the presets, we can then consider adding `(legacy)` presets.
> 
> How different are the updated versions? I didn't think the changes were that noticeable (just from reading prs, not from eating them)

Experimental Dark is quite noticeably different - the background is brighter and less saturated, and inactive tabs are gray instead of blue.

The change to TW Dark is much smaller - borders are slightly brighter gray, and I just noticed the change to the block palette background was incorrect and should be reverted.
